shared/chat/conversation/messages/wrapper/exploding-height-retainer/index.desktop.js

```diff
@@ -4,40 +4,73 @@ import * as ReactDOM from 'react-dom'
 import {resolveRootAsURL} from '../../../../../desktop/app/resolve-root.desktop'
 import {urlsToImgSet} from '../../../../../common-adapters/icon.desktop'
 import {Box, ConnectedUsernames, Text} from '../../../../../common-adapters'
-import {collapseStyles, globalColors, styleSheetCreate} from '../../../../../styles'
+import {
+  collapseStyles,
+  glamorous,
+  globalColors,
+  platformStyles,
+  styleSheetCreate,
+} from '../../../../../styles'
 import type {Props} from '.'
 
 const explodedIllustration = resolveRootAsURL('../images/icons/pattern-ashes-desktop-400-68.png')
 const explodedIllustrationUrl = urlsToImgSet({'68': explodedIllustration}, 68)
 
-const messageHeights = {}
+const copyChildren = children =>
+  React.Children.map(children, child => (child ? React.cloneElement(child) : child))
+
+const animationDuration = 1.5
+
+const retainedHeights = {}
 
 type State = {
-  height: ?number,
+  animating: boolean,
+  children: ?React.Node,
```

Comment:
why do we need to clone the children?

Reply:
As soon as the message explodes the text is deleted from the store; cloning the children is so the text doesn't disappear while the animation is running